vms/proposervm/state_summary.go

```diff
@@ -62,5 +56,8 @@ func (s *stateSummary) Accept(ctx context.Context) (block.StateSyncMode, error)
 	// innerSummary.Accept may fail with the proposerVM block and index already
 	// updated. The error would be treated as fatal and the chain would then be
 	// repaired upon the VM restart.
+	// After the inner summary is accepted, the engine transitions to bootstrapping
+	// and SetState is responsible for re-aligning the ProposerVM to the height reported
+	// by the inner VM after handling state sync.
```

Comment:
If we are transitioning out of state sync, then we already re-align the proposervm to the inner vm during that exit: https://github.com/ava-labs/avalanchego/blob/66490e40149bb64cdaf750c88e6fb763e7595bc8/vms/proposervm/vm.go#L272